src/Angor/Client/Pages/Invest.razor

```diff
@@ -413,22 +415,58 @@ else
                 }
             }
         }
+        
         await CheckIfSeederTimeHasPassed();
-
         UpdateStagesBreakdown(new ChangeEventArgs { Value = Investment.InvestmentAmount });
     }
 
+    private async Task CheckForExistingProjectAndCheckPassword()
+    {
+        if (!passwordComponent.HasPassword())
+        {
+            passwordComponent.ShowPassword(CheckForExistingInvestedProjects);
+        }
+        else
+        {
+            await CheckForExistingInvestedProjects();
+        }
+    }
+    
+    private async Task CheckForExistingInvestedProjects()
+    {
+        if (project != null)
+        {
+            
+            var words = await passwordComponent.GetWalletAsync();
+            var investorPubKey = _DerivationOperations.DeriveInvestorKey(words, project.ProjectInfo.FounderKey);
+
+            var isAlreadyInvested = await IsInvestorAlreadyInvestedAsync(project.ProjectInfo.ProjectIdentifier, investorPubKey);
+            if (isAlreadyInvested)
+            {
+                notificationComponent.ShowErrorMessage("You are already invested in this project.");
+                NavigationManager.NavigateTo($"/view/{ProjectId}");
```

Comment:
instead of throwing back to the view page, maybe just change the ui to say "already invested"